inc/class-base-command.php

```diff
@@ -0,0 +1,153 @@
+ * process plugins or themes, and generate tabular output with various filters.
+ *
+ * @package NetworkPluginAuditor\WPCLI
+ */
+
+namespace NetworkPluginAuditor\WPCLI;
+
+use WP_CLI;
+use WP_Site;
+
+if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	/**
+	 * A base command for WP-CLI.
+	 */
+	abstract class Base_Command {
+		protected array $sites;
+
+		/**
+		 * Executes the command to retrieve and store all sites in a multisite installation.
+		 *
+		 * @param array $args Positional arguments passed to the command.
+		 * @param array $assoc_args Associative arguments passed to the command.
+		 *
+		 * @return void
+		 */
+		public function run( array $args, array $assoc_args ) : void {
+			if ( ! is_multisite() ) {
+				WP_CLI::error( 'This command can only be run on a multisite installation.' );
+
+				return;
+			}
+
+			$sites = get_sites( [ 'number' => 0 ] ); // Get all sites
+			if ( empty( $sites ) ) {
+				WP_CLI::error( 'No sites found in the network.' );
+
+				return;
+			}
+
+			$this->sites = $sites;
+		}
+
+		/**
+		 * Generates a filtered and sorted table based on active plugins or themes across a multisite installation.
+		 *
+		 * @param array  $list A list of plugins or themes to evaluate for active status.
+		 * @param array  $assoc_args An associative array of filters and sorting criteria, including:
+		 *                          - 'order-by': Sort the table by 'name' or 'active-sites'.
+		 *                          - 'min-active-sites': Minimum number of active sites required for inclusion.
+		 *                          - 'max-active-sites': Maximum number of active sites allowed for inclusion.
+		 *                          - 'site-id': Filter by a specific site ID.
+		 * @param string $column_name The name of the primary column to display in the table.
+		 *
+		 * @return array A filtered and sorted array of data, where each entry contains:
+		 *               - The primary column key with the plugin/theme name.
+		 *               - 'Active Sites': The count of sites where the plugin/theme is active.
+		 *               - 'Site IDs': A comma-separated list of site IDs where the plugin/theme is active.
+		 */
+		public function get_table( array $list, array $assoc_args, string $column_name ) : array {
+			$active_counts = array_fill_keys( $list, 0 );
+			$active_sites = array_fill_keys( $list, [] );
+
+			// Count active themes per site and store site IDs
+			foreach ( $this->sites as $site ) {
+				$plugins_or_themes = $this->before_count( $site );
+
+				foreach ( $plugins_or_themes as $plugin_or_theme ) {
+					if ( isset( $active_counts[ $plugin_or_theme ] ) ) {
+						$active_counts[ $plugin_or_theme ] ++;
+					} else {
+						$active_counts[ $plugin_or_theme ] = 1;
+					}
+
+					if ( isset( $active_sites[ $plugin_or_theme ] ) ) {
+						$active_sites[ $plugin_or_theme ][] = $site->blog_id;
+					} else {
+						$active_sites[ $plugin_or_theme ] = [ $site->blog_id ];
+					}
```

Comment:
You _can_ actually do this without warnings:

```suggestion
					$active_sites[ $plugin_or_theme ][] = $site->blog_id;
```

Reply:
This works
```
$active_counts[ $plugin_or_theme ] = ( $active_counts[ $plugin_or_theme ] ?? 0 ) + 1;
$active_sites[ $plugin_or_theme ][] = $site->blog_id;
```